### PR TITLE
test: add unit tests for agent_debugger_sdk/telemetry (zero coverage)

### LINKED ISSUE #304: test: add unit tests for agent_debugger_sdk/telemetry (zero coverage)
## Problem

\`agent_debugger_sdk/telemetry/__init__.py\` (40 statements) has 0% test coverage, confirmed via:

\`\`\`
.venv-ci/bin/python -m pytest -q --cov=agent_debugger_sdk --cov-report=term-missing
\`\`\`

## Scope

Cover the OTel-optional telemetry shim:

- \`init_telemetry()\` — ImportError fallback path (opentelemetry not installed → no-op, logs info)
- \`init_telemetry()\` — otlp exporter branch with/without endpoint, and its own nested ImportError fallback to console exporter (requires opentelemetry installed to exercise fully; otherwise assert the no-op path only)
- \`get_tracer()\` — ImportError fallback returns \`_NoOpTracer\`
- \`_NoOpTracer.start_as_current_span()\` / \`_NoOpContextManager.__enter__\` / \`__exit__\` — no-op contract
- \`is_telemetry_enabled()\` — reflects \`_initialized\` state before/after \`init_telemetry()\`

No external OTel dependency required for the primary no-op paths (which is what this env actually exercises).

## Acceptance

- New \`tests/test_telemetry.py\` (or similar) covering the paths above
- \`ruff check .\` clean
- \`pytest -q\` green

### PR DESCRIPTION
**Pull request by acailic**
## Summary
- Adds `tests/test_telemetry.py` covering `agent_debugger_sdk/telemetry/__init__.py`, previously at 0% coverage
- Covers no-op fallback paths (opentelemetry not installed — the only branches this environment can naturally reach) plus the success paths (console export, otlp export with/without the otlp exporter package, `get_tracer` delegation) via injected fake modules in `sys.modules`
- Module coverage: 0% → 100%

Closes #304

## Test plan
- [x] `ruff check .` clean
- [x] `pytest -q tests/test_telemetry.py --cov=agent_debugger_sdk.telemetry --cov-report=term-missing` → 12 passed, 100% coverage
- [x] Full suite: `pytest -q` → 3035 passed, 19 skipped, 0 failed (up from 3023 passed on main)

🤖 Generated with [Amplifier](https://github.com/microsoft/amplifier)

### CODE REVIEW

File: tests/test_telemetry.py [hunk 1]
```diff
@@ -0,0 +1,207 @@
...
+        return_value=MagicMock(add_span_processor=MagicMock())
+    )
+
+    export_mod = ModuleType("opentelemetry.sdk.trace.export")
+    export_mod.BatchSpanProcessor = MagicMock()
+    export_mod.ConsoleSpanExporter = MagicMock()
+
+    sdk_mod = ModuleType("opentelemetry.sdk")
+    sdk_mod.trace = sdk_trace_mod
+
```
[the remark this answers is not in the document]
Comment (acailic) [reply tests/test_telemetry.py]: Fixed — fake modules now get `__path__` (marked as packages) and are wired onto their parents (`otel_mod.sdk = sdk_mod`, etc.). Verified this specific concern by running the suite against a venv with real opentelemetry installed: without `__path__` on the fake `opentelemetry` stub, an unrelated test (the otlp-fallback one) fell through to the real `opentelemetry.exporter` namespace package on disk and crashed with `AttributeError: module 'opentelemetry' has no attribute '__path__'`. Empty `__path__` on the fakes now makes unregistered submodule imports fail cleanly instead.

File: tests/test_telemetry.py [hunk 2]
```diff
@@ -0,0 +1,207 @@
...
+    """is_telemetry_enabled should reflect the module-level _initialized flag."""
+    import agent_debugger_sdk.telemetry as telemetry
+
+    telemetry._initialized = False
+    assert telemetry.is_telemetry_enabled() is False
+
+
+def test_init_telemetry_noop_when_opentelemetry_not_installed(caplog):
+    """init_telemetry should no-op and log when opentelemetry-sdk is unavailable."""
+    import agent_debugger_sdk.telemetry as telemetry
```
[the remark this answers is not in the document]
Comment (acailic) [reply tests/test_telemetry.py]: Confirmed and fixed. Ran this test against a venv with opentelemetry actually installed — it failed as predicted. Added a `no_opentelemetry` fixture that shadows `sys.modules['opentelemetry']` with a bare, path-less dummy module, forcing the ImportError branch deterministically. Test now passes in both a no-otel venv and a real-otel venv.

File: tests/test_telemetry.py [hunk 3]
```diff
@@ -0,0 +1,207 @@
...
+    assert result is None
+    assert telemetry.is_telemetry_enabled() is False
+    assert "opentelemetry-sdk not installed" in caplog.text
+
+
+def test_init_telemetry_accepts_otlp_args_without_crashing():
+    """init_telemetry should not raise when called with an otlp exporter/endpoint,
+    even though opentelemetry is not installed (falls back to the no-op path)."""
+    import agent_debugger_sdk.telemetry as telemetry
+
```
[the remark this answers is not in the document]
Comment (acailic) [reply tests/test_telemetry.py]: Confirmed and fixed — same `no_opentelemetry` fixture applied here. Verified failing before the fix (against a venv with opentelemetry installed) and passing after.

File: tests/test_telemetry.py [hunk 4]
```diff
@@ -0,0 +1,207 @@
...
+        exporter="otlp",
+    )
+
+    assert telemetry.is_telemetry_enabled() is False
+
+
+def test_get_tracer_returns_noop_tracer_when_opentelemetry_not_installed():
+    """get_tracer should fall back to _NoOpTracer when opentelemetry is unavailable."""
+    from agent_debugger_sdk.telemetry import _NoOpTracer, get_tracer
+
```
[the remark this answers is not in the document]
Comment (acailic) [reply tests/test_telemetry.py]: Confirmed and fixed — same `no_opentelemetry` fixture applied here. Verified failing before the fix (against a venv with opentelemetry installed) and passing after.
